### ISSUE TITLE
Padding to original bounds extends the array

### ISSUE THREAD

**Issue by davidbrochart** [corteva/rioxarray#155]
#### Code Sample, a copy-pastable example if possible

```python
import rioxarray
import xarray

xds = xarray.open_dataarray("test/test_data/input/MODIS_ARRAY.nc")
# xds.shape is (200, 200)
bounds = xds.rio.bounds()
xdsc = xds.rio.pad_box(
    minx=bounds[0],
    miny=bounds[1],
    maxx=bounds[2],
    maxy=bounds[3],
)
# xdsc.shape is (201, 200)
```
#### Problem description

I think we shouldn't include the half-pixel [here](https://github.com/corteva/rioxarray/blob/0006f067f8fadc967060fa4889c02eaa2ec2d9ff/rioxarray/rioxarray.py#L1365-L1372).

#### Expected Output

The shape of the padded array should remain the same.

**Comment by snowman2** [comment]
Does `pad_xy` behave as expected?

**Comment by davidbrochart** [comment]
Yes, but I guess it's a question of definition of the bounding box. In `pad_box`, do we expect a box that is relative to the "pixel boundaries" or the pixel centers (which correspond to the array coordinates)?

**Comment by snowman2** [comment]
> do we expect a box that is relative to the "pixel boundaries" 

I would assume that would be the behavior. What are your thoughts?

**Comment by davidbrochart** [comment]
I don't know, but it has to be consistent with `clip_box`. It looks like `clip_box` expects a box relative to the "pixel centers" (it extends it with a half-pixel to the "pixel boundaries"), so in this case `pad_box` behaves similarly (although `clip_box` uses `rasterio.windows.from_bounds`).
Calling `pad_box` with the `rio.bounds()` (which include the half-pixel) can lead to a random behavior because of floating point approximations, so it's not a good idea. It's better in that case to call it with the first/last coordinates.

**Comment by snowman2** [comment]
One idea I had was to reduce the buffer by multiplying the cell size by 0.9999 and see if that helps with the floating point precision issues.

**Comment by davidbrochart** [comment]
I'm not sure, it would be kind of a hack. I think the current behavior is OK, we might just have to document it a bit more.

**Comment by snowman2** [comment]
That sounds good to me :+1:
